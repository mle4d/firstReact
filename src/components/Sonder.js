import React from 'react';

export default function Sonder({ Name, Birthday, Lifestyle }) {
  const Sonder = {
    Name,
    Birthday,
    Lifestyle
  };
  const fields = Object.keys(Sonder)
    .map(key => {
      return (
        <>
          <dt>{key}</dt>
          <dd>{Sonder[key]}</dd>
        </>
      );
    });
  return (
    <dl>
      {fields}
    </dl>
  );
}
