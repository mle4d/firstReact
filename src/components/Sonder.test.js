import React from 'react';
import { shallow } from 'enzyme';
import Sonder from './Sonder';

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(<Sonder />);
    expect(wrapper).toMatchSnapshot();
  });
});
