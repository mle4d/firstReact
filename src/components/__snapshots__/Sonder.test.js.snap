// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<dl>
  <dt>
    Name
  </dt>
  <dd />
  <dt>
    Birthday
  </dt>
  <dd />
  <dt>
    Lifestyle
  </dt>
  <dd />
</dl>
`;
